### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by NatalyKrvch**
[DEMO LINK](https://NatalyKrvch.github.io/react_list-of-goods-js/)

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -14,45 +19,93 @@ export const goodsFromServer = [
...
+            }}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul>
+        {visibleGoods.map(good => (
+          <li data-cy="Good">{good}</li>
```
Comment (Esceype) [inline src/App.jsx:106]:
```suggestion
          <li data-cy="Good">{good}</li>
```

Add `key` attribute

**Review by Esceype — CHANGES_REQUESTED** [review]
Great job!
Just one small but very important improvement.

File: src/App.jsx (hunk 1 above)
Comment (Viktor-Kost) [reply src/App.jsx:106]: you can use `good` as a key

File: src/App.jsx [hunk 2]
```diff
@@ -14,45 +19,93 @@ export const goodsFromServer = [
...
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
```
Comment (Viktor-Kost) [inline src/App.jsx:49]:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

File: src/App.jsx [hunk 3]
```diff
@@ -14,45 +19,93 @@ export const goodsFromServer = [
...
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortType, isReverse);
+  const showResetBtn = sortType || isReverse;
```
Comment (Viktor-Kost) [inline src/App.jsx:52]: it's ok, yet since you have such an easy condition you can use it directly at jsx

File: src/App.jsx [hunk 4]
```diff
@@ -14,45 +19,93 @@ export const goodsFromServer = [
...
+        </button>
+
+        {showResetBtn && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortType('');
+              setIsReverse(false);
+            }}
```
Comment (Viktor-Kost) [inline src/App.jsx:97]: it's a good practice to move such handlers to an own function

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/App.jsx [hunk 5]
```diff
@@ -14,45 +19,94 @@ export const goodsFromServer = [
...
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const getPreparedGoods = (goods, sortType, isReverse) => {
```
Comment (Viktor-Kost) [inline src/App.jsx:22]:
```suggestion
const getPreparedGoods = (goods, sortType, isReversed) => {
```

File: src/App.jsx [hunk 6]
```diff
@@ -14,45 +19,94 @@ export const goodsFromServer = [
...
+        case LENGTH_SORT:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
```
Comment (Viktor-Kost) [inline src/App.jsx:40]:
```suggestion
  if (isReversed) {
```

**Review by Viktor-Kost — APPROVED** [review]
Nice!

**Pull request closed without merging**